ambry-messageformat/src/main/java/com.github.ambry.messageformat/MessageSievingInputStream.java

```diff
@@ -152,81 +157,67 @@ public boolean hasInvalidMessages() {
-      ByteBuffer headerVersion = ByteBuffer.wrap(headerVersionInBytes);
-      short version = headerVersion.getShort();
-      if (!isValidHeaderVersion(version)) {
-        throw new MessageFormatException("Header version not supported " + version,
-            MessageFormatErrorCodes.Data_Corrupt);
-      }
-      ByteBuffer headerBuffer = ByteBuffer.allocate(getHeaderSizeForVersion(version));
-      headerBuffer.putShort(version);
-      byteArrayInputStream.read(headerBuffer.array(), 2, headerBuffer.capacity() - 2);
-      headerBuffer.position(headerBuffer.capacity());
-      headerBuffer.flip();
-      MessageHeader_Format header = getMessageHeader(version, headerBuffer);
-      StoreKey storeKey = storeKeyFactory.getStoreKey(new DataInputStream(byteArrayInputStream));
-      if (header.isPutRecord()) {
-        encryptionKey = header.hasEncryptionKeyRecord() ? deserializeBlobEncryptionKey(byteArrayInputStream) : null;
-        props = deserializeBlobProperties(byteArrayInputStream);
-        metadata = deserializeUserMetadata(byteArrayInputStream);
-        blobData = deserializeBlob(byteArrayInputStream);
-      } else {
-        throw new IllegalStateException("Message cannot be a deleted record ");
-      }
-      if (byteArrayInputStream.available() != 0) {
-        logger.error("Parsed message size {} is not equivalent to the size in message info {}",
-            (availableBeforeParsing + byteArrayInputStream.available()), availableBeforeParsing);
-      } else {
-        logger.trace("Message Successfully read");
-        logger.trace(
-            "Header - version {} Message Size {} Starting offset of the blob {} BlobEncryptionKeyRecord {} BlobPropertiesRelativeOffset {}"
-                + " UserMetadataRelativeOffset {} DataRelativeOffset {} UpdateRecordRelativeOffset {} Crc {}",
-            header.getVersion(), header.getMessageSize(), currentOffset,
-            header.getBlobEncryptionKeyRecordRelativeOffset(), header.getBlobPropertiesRecordRelativeOffset(),
-            header.getUserMetadataRecordRelativeOffset(), header.getBlobRecordRelativeOffset(),
-            header.getUpdateRecordRelativeOffset(), header.getCrc());
-        logger.trace("Id {} Encryption Key -size {} Blob Properties - blobSize {} Metadata - size {} Blob - size {} ",
-            storeKey.getID(), encryptionKey == null ? 0 : encryptionKey.capacity(), props.getBlobSize(),
-            metadata.capacity(), blobData.getSize());
-        if (msgInfo.getStoreKey().equals(storeKey)) {
-          isValid = true;
+  private void sieve(Message inMsg, OutputStream sievedOutputStream, int msgOffset) throws IOException {
+    if (transformers == null || transformers.isEmpty()) {
+      // Write the message without any transformations.
+      sievedMessageInfoList.add(inMsg.getMessageInfo());
+      Utils.transferBytes(inMsg.getStream(), sievedOutputStream, inMsg.getMessageInfo().getSize());
```

Comment:
I _think_ you can avoid this by plugging the input stream into the outputstream but im not entirely sure - can handle this later, but could you just add a comment?

Reply:
With the `SequenceInputStream`, this may no longer be needed